lib/bpl/transformation/shadowing.rb

```diff
@@ -26,6 +26,8 @@ def shadow_copy(node)
         shadow
       end
 
+      def shadow_expr_eq(expr) "#{expr} == #{shadow_copy(expr)}" end
```

Comment:
No reason other than my attachment to strong typing :)

I have since realized that I'm not sure shadow_copy is properly descending
into subexpressions recursively, which may lead to the same issue arising
again further down the line.

On Fri, 18 Sep 2015 at 19:36 Michael Emmi <notifications@github.com> wrote:

> In lib/bpl/transformation/shadowing.rb
> <https://github.com/michael-emmi/bam-bam-boogieman/pull/3#discussion_r39880568>
> :
>
> > @@ -26,6 +26,8 @@ def shadow_copy(node)
> >          shadow
> >        end
> >
> > +      def shadow_expr_eq(expr) "#{expr} == #{shadow_copy(expr)}" end
>
> Any good reason not to just define shadow_eq this way, rather than having
> both shadow_eq and shadow_expr_eq?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/michael-emmi/bam-bam-boogieman/pull/3/files#r39880568>
> .
>